lib/dry/auto_inject/injector.rb

```diff
@@ -2,22 +2,36 @@
 
 module Dry
   module AutoInject
-    class Injector
+    class Injector < BasicObject
       # @api private
       attr_reader :container
 
       # @api private
       attr_reader :strategy
 
       # @api private
-      def initialize(container, strategy)
+      attr_reader :builder
+
+      # @api private
+      def initialize(container, strategy, builder:)
         @container = container
         @strategy = strategy
+        @builder = builder
       end
 
       def [](*dependency_names)
         strategy.new(container, *dependency_names)
       end
+
+      def respond_to?(name, include_private = false)
+        Injector.instance_methods.include?(name) || builder.respond_to?(name)
```

Comment:
This still isn't right lol

```ruby
def respond_to?(name, include_private = false)
  method_finder = include_private ? :methods : :public_methods
  # Note that sending `public_instance_methods` to the class explictly
  # rather than self will break this methods behaviour on subclasses
  __send__(method_finder).include?(name) || builder.respond_to?(name)
end
```